pyro/distributions/kl.py

```diff
@@ -43,4 +43,16 @@ def _kl_independent_mvn(p, q):
     raise NotImplementedError
 
 
+# TODO: move upstream
+@register_kl(TransformedDistribution, TransformedDistribution)
+def _kl_transformed_transformed(p, q):
+    if p.transforms != q.transforms:
+        raise NotImplementedError
+    if p.event_shape != q.event_shape:
+        raise NotImplementedError
+    extra_event_dim = len(p.event_shape) - len(p.base_dist.event_shape)
```

Comment:
The upstream version `extra_event_dim = len(p.event_shape)` seems incorrect to me. However, I am not sure if this change will cover all the cases. Discussions are welcomed!